src/ui/MultiSelect/MultiSelect.tsx

```diff
@@ -0,0 +1,137 @@
+import { useEffect, useRef, useState } from 'react'
+import { IoClose } from 'react-icons/io5'
+import { CgChevronDown, CgChevronUp } from 'react-icons/cg'
+
+import {
+  InputBox,
+  Tag,
+  TagsWrapper,
+  MirroredIcon,
+  StyledOption,
+  OptionsWrapper,
+} from './MultiSelect.styles'
+import InputSearch from '../InputSearch/InputSearch'
+import useOutsideClick from 'hooks/useClickOutside'
+
+interface IServiceOptions {
+  key: string
+  name: string
+}
+
+interface MultiSelectProps {
+  options: IServiceOptions[]
+  onChange: (selectedServices: IServiceOptions[]) => void
+  placeholder: string
+}
+
+const MultiSelect: React.FC<MultiSelectProps> = ({
+  options,
+  onChange,
+  placeholder,
+}) => {
+  const [isOpen, setIsOpen] = useState(false)
+  const [searchTerm, setSearchTerm] = useState<string>('')
+  const [suggestions, setSuggestions] = useState<IServiceOptions[]>(options)
+  const [selectedServices, setSelectedServices] = useState<IServiceOptions[]>(
+    []
+  )
+  const [selectedServicesSet, setSelectedServicesSet] = useState(new Set())
+  const refOptions = useOutsideClick(() => setIsOpen(false))
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  useEffect(() => {
+    if (searchTerm.trim() === '') {
+      setSuggestions(options)
+      return
+    }
+
+    const filteredOptions = options.filter((element) =>
+      element.name.toLowerCase().includes(searchTerm.toLowerCase())
+    )
+
+    console.log(searchTerm)
+    console.log(filteredOptions)
+    setSuggestions(filteredOptions)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [searchTerm])
+
+  const handleAddService = (service: IServiceOptions) => {
+    const updatedSelectedServices = [...selectedServices, service]
+    setSelectedServices(updatedSelectedServices)
+    setSelectedServicesSet(new Set([...selectedServicesSet, service.key]))
+    onChange(updatedSelectedServices)
+    if (inputRef.current !== null) {
+      inputRef.current.focus()
+    }
+  }
+
+  const handleRemoveService = (service: IServiceOptions) => {
+    const filteredArray = selectedServices.filter(
+      (ele) => ele.key !== service.key
+    )
+    setSelectedServices(filteredArray)
+    const updatedServices = new Set(selectedServicesSet)
+    updatedServices.delete(service.key)
+    setSelectedServicesSet(updatedServices)
+    onChange(filteredArray)
+  }
+
+  const handleIsOpen = () => {
```

Comment:
remove this function and send directly the SetIsOpen to onclick event

Reply:
done